### ISSUE TITLE
excludeDependencies?

### ISSUE THREAD

**Issue by jay8t6** [Polymer/polymer-build#36]
Can we include excludeDependencies option as part of the build? or is there a simpler way to exclude files from the build?

**Comment by web-padawan** [comment]
seems to be a duplicate of #31

**Comment by FredKSchott** [comment]
@web-padawan not exactly, I believe you're interested in excluding files from bundles in #31. @jay8t6 seems to be interested in excluding files entirely from the final builds.

@jay8t6 can you explain your use case here? I'm not sure if I understand the need for excluding files if the analyzer detected them being loaded in your application.

**Comment by jay8t6** [comment]
@FredKSchott say as part of the build, I want to include some images thats on my bower dependencies by doing bower_components/**/images/* in includeDependencies, it loads in all the images folder, it works great. But it also loads images that I don't need, for example the images from /app-layout/site/images folder. which takes up about 7mb of my total build size (the images sizes on app-layout repo is huge by the way, https://github.com/PolymerElements/app-layout/tree/master/site/images).

So, I was hoping I could do something like:

"excludeDependencies": [
'bower_components/app-layout/site/images/*'
]

Makes sense?

**Comment by FredKSchott** [comment]
Gotcha. I'd reccomend not being so broad in your definition of `bower_components/**/images/*`, and instead including only the `images` directories that you're interested in.

But if that doesn't work, we *should* support "!" globs, meaning you could do something like this:

```
"includeDependencies": [
"bower_components/**/images/*", 
"!bower_components/app-layout/site/images/*"
]
```

Please let us know if that doesn't work.

**Comment by jay8t6** [comment]
Oh nice... I was broader because, I have multiple bower dependencies that may or may not include images, I dont wanna add each and every one of them here. However, I think the ! will do what I want, because thats the only component that's increasing my build size. Thanks for the solution, I will give that a try!

**Comment by jay8t6** [comment]
@FredKSchott I tried the '!' approach, it doesn't seem to work for me. Has it been implemented or you said you will support that in the future?

**Comment by FredKSchott** [comment]
I'm surprised to hear that, we pass the globs to [vinyl-fs](https://github.com/gulpjs/vinyl-fs#srcglobs-options) which [explicitly supports this](https://github.com/gulpjs/vinyl-fs#srcglobs-options). Can you share your `includeDependencies` blob? Does anything in their README help you?

If not I'll take another look for you.

**Comment by FredKSchott** [comment]
Ah, the problem is that we're resolving the paths incorrectly, resulting in something like `/Users/johndoe/Code/temp/!bower_components/**/*.md` instead of `!/Users/johndoe/Code/temp/bower_components/**/*.md` 

Funnily enough this was already fixed and tested in polymer-project-config, which we're in the process of integrating now. This will be fixed when #43 is merged in.
